pandas/tests/series/test_constructors.py

```diff
@@ -2216,7 +2216,16 @@ def test_np_string_array_object_cast(self, data):
         arr = np.array(data, dtype=StringDType())
         res = Series(arr)
         assert res.dtype == np.object_
-        assert (res == data).all()
+
+        if data[-1] is np.nan:
+            # as of GH#62522 the comparison op for `res==data` casts data
+            #  using sanitize_array, which casts to 'str' dtype, which does not
+            #  consider string 'nan' to be equal to np.nan,
+            #  (which apparently numpy does?  weird.)
```

Comment:
I suppose if/when we support `np.StringDtype`, we can treat the data as masked and then we hopefully won't be comparing `'nan'` to `np.nan`?